ci/windows/build_common_python.psm1

```diff
@@ -0,0 +1,288 @@
+function Get-LatestPythonPatchVersionFromPyEnvWin {
+    <#
+    .SYNOPSIS
+        Resolves the latest patch version for a given Python major.minor (e.g.
+        '3.12') by parsing `pyenv install --list` output on Windows (pyenv-win).
+    .PARAMETER Version
+        A string in the form 'M.m' (e.g., '3.10', '3.11', '3.12').
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory, Position = 0)]
+        [ValidatePattern('^\d+\.\d+$')]
+        [string]$Version
```

Comment:
It's not important, nor does it matter greatly, but there is a canonical version type. https://learn.microsoft.com/en-us/dotnet/api/system.version?view=net-9.0